protocol/message_test.go

```diff
@@ -68,6 +68,22 @@ func TestSerializeANormalMessage(t *testing.T) {
 	assert.Equal(t, strings.SplitN(aNormalMessageNoExpires, "\n", 2)[0], msg.Metadata())
 }
 
+func TestCorrelationID(t *testing.T) {
```

Comment:
you could add 2 other simple tests to test the error cases: 
unmarshalling json
non-existing Correlation-Id